### PR TITLE
Add Script to Analyze Winner's Score Differentials

### PR DESCRIPTION
**Pull request by ncteisen**
Given a winner from a year, what is the closes game it won to earn its championship? Let's rank that number.

### CODE REVIEW

**Comment by ncteisen** [comment]
Output:

```
$ ./most_dominant.py data/*

2009 - North Carolina (1): 12
2000 - Michigan State (1): 11
2001 -           Duke (1): 10
2012 -       Kentucky (1):  8
2002 -       Maryland (1):  8
2007 -        Florida (1):  7
1996 -       Kentucky (1):  7
1993 - North Carolina (1):  6
2015 -           Duke (1):  5
2014 -    Connecticut (7):  5
2013 -     Louisville (1):  4
2006 -        Florida (3):  4
1994 -       Arkansas (1):  4
2016 -      Villanova (2):  3
1999 -    Connecticut (1):  3
1997 -        Arizona (4):  3
1988 -         Kansas (6):  3
1986 -     Louisville (2):  3
2010 -           Duke (1):  2
2008 -         Kansas (1):  2
1991 -           Duke (2):  2
1990 -           UNLV (1):  2
1985 -      Villanova (8):  2
2017 - North Carolina (1):  1
2011 -    Connecticut (3):  1
2005 - North Carolina (1):  1
2004 -    Connecticut (2):  1
2003 -       Syracuse (3):  1
1998 -       Kentucky (2):  1
1995 -           UCLA (1):  1
1992 -           Duke (1):  1
1989 -       Michigan (3):  1
1987 -        Indiana (1):  1
```

This means that in 2009, UNC won their **closest** game of the tournament by 12 points. Let's dig into that season:

```
./trace_winner.py data/2009.json

2009 - Winner: North Carolina:
	In the   Round of 64, they beat              Radford* 101 -  58  (+43)
	In the   Round of 32, they beat                   LSU  84 -  70  (+14)
	In the Sweet Sixteen, they beat               Gonzaga  98 -  77  (+21)
	In the   Elite Eight, they beat              Oklahoma  72 -  60  (+12)
	In the    Final Four, they beat             Villanova  83 -  69  (+14)
	In the  Championship, they beat          Michigan St.  89 -  72  (+17)
```

*I had to add Radford in manually. It seems to be missing from the dataset

File: most_dominant.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env python3
+"""Given a winner, what was the closest game they played in the tournament?"""
+
+import sys
+
+from utils import sum_for_game, get_flattened_games
+
+def get_winner(year, game_years):
```
Comment (danvk) [inline most_dominant.py:8]: Your `game_years` construct is working against you here. The year and winner are readily available if you go directly from the JSON files. (The year is always `data['year']` and the championship is always `data['finalfour'][2]`).

File: trace_winner.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+_ROUNDS_NAMES = [
+    (64, "Round of 64"),
+    (32, "Round of 32"),
+    (16, "Sweet Sixteen"),
+    (8,  "Elite Eight"),
+    (4,  "Final Four"),
+    (2,  "Championship"),
+]
+
+def get_winner(year, game_years):
```
Comment (danvk) [inline trace_winner.py:18]: Pull this into `utils`.

**Comment by danvk** [comment]
I'll try to figure out why Radford wasn't picked up. I'll be there are other anomalies in the bracket data that we've missed.

**Pull request closed without merging**